src/prefect/server/services/task_run_recorder.py

```diff
@@ -25,13 +25,33 @@
     MessageHandler,
     create_consumer,
 )
+from prefect.server.utilities.messaging.memory import (
+    METRICS,
+    _metrics_lock,  # type: ignore
+)
 
 if TYPE_CHECKING:
     import logging
 
 logger: "logging.Logger" = get_logger(__name__)
 
 
+async def log_metrics_periodically(interval: float = 2.0) -> None:
+    while True:
+        await asyncio.sleep(interval)
+        async with _metrics_lock:
+            for topic, data in METRICS.items():
+                depth = data["published"] - data["consumed"]
+                logger.debug(
+                    "Topic=%r | published=%d consumed=%d retried=%d depth=%d",
+                    topic,
+                    data["published"],
+                    data["consumed"],
+                    data["retried"],
+                    depth,
+                )
```

Comment:
Could this be a utility in `prefect.server.utilities.messaging.memory` so we don't need to import `_metrics_lock` directly?

Reply:
good call